ignite/engine/__init__.py

```diff
@@ -76,6 +76,7 @@ def create_supervised_evaluator(model, metrics={},
         model.to(device)
 
     def _inference(engine, batch):
+        engine.state.output = None
```

Comment:
@IlyaOvodov here `engine` is evaluator and not the trainer. Is it OK ?

Reply:
Yes, see my message above. Traines output is just loss value and it does not produse such a problem. It's a pity that standard trainer does not return pred_y and y, it forces to use customizes one. But I don't see a way to change it without pain for existing users :)

Reply:
I see. Maybe we can introduce an optional `output_transform` which by default does 
```python
def default_output_transform(x, y, y_pred, loss):
    return loss.item()
```
And user could override the function...

Reply:
@vfdev-5 ... and pass it as parameter to create_supervised_trainer? Sounds good...
But I think make 2 functions `output_loss` and `output_y_pred_y` and:
```python
def create_supervised_trainer(... , output_transform = output_loss)
def create_supervised_evaluator(... , output_transform = output_y_pred_y)
```
- it is more meaningful, IMHO